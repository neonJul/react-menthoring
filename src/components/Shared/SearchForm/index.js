import React from 'react';
import Button from '../Button';

import './index.css';

class Header extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            inputValue: ''
        }
    }

    handleChange = (event) => {
        this.setState({
            inputValue: event.target.value
        })
    }

    handleSubmit = (event) => {
        event.preventDefault();
    }

    render() {
        return (
            <form className="search-form" onSubmit={this.handleSubmit}>
                <input type="text" value={this.state.inputValue} onChange={this.handleChange}/>
                <Button name="Search"/>
            </form>
        )
    }
};

export default Header;