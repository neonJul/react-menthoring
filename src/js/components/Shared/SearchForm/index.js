import React from 'react';
import Button from '../Button';

import './index.css';

class Header extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            inputValue: ''
        }
    }

    handleChange = (event) => {
        this.setState({
            inputValue: event.target.value
        })
    }

    handleSubmit = (event) => {
        event.preventDefault();
    }

    render() {
        return (
            <form className="search-form">
                <input type="text" value={this.state.inputValue} onChange={this.handleChange}/>
                <Button name="Search" onClick={this.handleSubmit}/>
            </form>
        )
    }
};

export default Header;