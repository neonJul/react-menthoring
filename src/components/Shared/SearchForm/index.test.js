import React from 'react';
import renderer from 'react-test-renderer';

import SearchForm from './index';

it('renders correctly', () => {
  const tree = renderer.create(<SearchForm />).toJSON();
  expect(tree).toMatchSnapshot();
});