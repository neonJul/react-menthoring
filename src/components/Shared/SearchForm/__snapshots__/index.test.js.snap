// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="search-form"
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    className="button"
  >
    Search
  </button>
</form>
`;
